core/templates/dev/head/components/forms/schema_editors/SchemaBasedListEditorDirective.js

```diff
@@ -135,12 +135,23 @@ oppia.directive('schemaBasedListEditor', [
             }
           };
 
+          var _deleteElementIfEmpty = function() {
+            for (var i = 0; i < $scope.localValue.length - 1; i++) {
+              if ($scope.localValue[i].length === 0) {
+                $scope.deleteElement(i);
+                i--;
+              }
+            }
+          };
+
           $scope.lastElementOnBlur = function() {
             _deleteLastElementIfUndefined();
+            _deleteElementIfEmpty();
             $scope.showAddItemButton();
           };
 
           $scope.showAddItemButton = function() {
+            _deleteElementIfEmpty();
```

Comment:
The name of this sounds like it's intended to delete only one element. Should we pass in the index of that element and have the function delete just that element (if it's empty)?